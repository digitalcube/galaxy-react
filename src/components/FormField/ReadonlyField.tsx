import React, { FC, ReactNode } from 'react';
import { useMemo } from 'react';
import { FaCopy } from 'react-icons/fa';
import { FormGroup, Input } from 'reactstrap';

export type ReadOnlyFormFieldProps = {
  value: string;
  icon: ReactNode;
  handleCopy: (value: string) => void;
  smaller?: boolean;
  className?: string;
};
export const ReadOnlyFormField: FC<ReadOnlyFormFieldProps> = ({
  value,
  icon,
  smaller,
  handleCopy,
  className: _className,
}) => {
  const className = useMemo(() => {
    const items = ['readonly'];
    if (!icon) items.push('no-icon');
    if (_className) items.push(_className);
    return items.filter(Boolean).join(' ');
  }, [_className, icon]);
  return (
    <FormGroup className={className}>
      <div className="form-group-inner position-relative">
        <Input
          type="text"
          className={`${smaller ? 'smaller ' : ''}py-0"`}
          value={value}
          readOnly
          onClick={() => handleCopy(value)}
        />
        <span className="icon position-absolute">
          <i aria-hidden="true">{icon}</i>
        </span>
        <span
          className="copy position-absolute js-copy-record"
          role="button"
          onClick={() => handleCopy(value)}
        >
          <i aria-hidden="true">
            <FaCopy className="d-block" />
          </i>
        </span>
      </div>
    </FormGroup>
  );
};
