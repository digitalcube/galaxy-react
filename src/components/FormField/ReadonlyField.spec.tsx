import { render } from '@testing-library/react';
import React from 'react';
import { Default as ReadOnlyFormField } from './ReadonlyField.stories';

describe('ReadOnlyFormField', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<ReadOnlyFormField {...(ReadOnlyFormField.args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot (with error massage)', () => {
    const { asFragment } = render(
      <ReadOnlyFormField
        {...(ReadOnlyFormField.args as any)}
        errorMessage="We donâ€™t recognize that email"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
