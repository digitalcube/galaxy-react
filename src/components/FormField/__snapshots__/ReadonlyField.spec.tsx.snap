// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadOnlyFormField to match snapshot (with error massage) 1`] = `
<DocumentFragment>
  <div
    class="readonly form-group"
  >
    <div
      class="form-group-inner position-relative"
    >
      <input
        class="py-0\\" form-control"
        readonly=""
        type="text"
        value="readonlyData"
      />
      <span
        class="icon position-absolute"
      >
        <i
          aria-hidden="true"
        >
          <svg
            class="d-block"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
        </i>
      </span>
      <span
        class="copy position-absolute js-copy-record"
        role="button"
      >
        <i
          aria-hidden="true"
        >
          <svg
            class="d-block"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
            />
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ReadOnlyFormField to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="readonly form-group"
  >
    <div
      class="form-group-inner position-relative"
    >
      <input
        class="py-0\\" form-control"
        readonly=""
        type="text"
        value="readonlyData"
      />
      <span
        class="icon position-absolute"
      >
        <i
          aria-hidden="true"
        >
          <svg
            class="d-block"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            />
          </svg>
        </i>
      </span>
      <span
        class="copy position-absolute js-copy-record"
        role="button"
      >
        <i
          aria-hidden="true"
        >
          <svg
            class="d-block"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
            />
          </svg>
        </i>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
